packages/editor/src/browser/editor.ts

```diff
@@ -20,6 +20,7 @@ export const TextEditorProvider = Symbol('TextEditorProvider');
 export type TextEditorProvider = (uri: URI) => Promise<TextEditor>;
 
 export interface TextEditorDocument extends lsp.TextDocument, Saveable, Disposable {
+    getLineContent(lineNumber: number): string;
```

Comment:
Could you add `lineCount` property? I think it is important for iteration.

Reply:
It's here `lsp.TextDocument.lineCount`